src/ConnectedMode.UnitTests/QualityProfiles/QualityProfileDownloaderTests.cs

```diff
@@ -0,0 +1,320 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.Binding;
+using SonarLint.VisualStudio.ConnectedMode.QualityProfiles;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Progress.Controller;
+using SonarLint.VisualStudio.TestInfrastructure;
+using SonarQube.Client;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.QualityProfiles
+{
+    [TestClass]
+    public class QualityProfileDownloaderTests
+    {
+        [TestMethod]
+        public async Task UpdateAsync_AllSupportedLanguageAreChecked()
+        {
+            var boundProject = CreateBoundProject();
+
+            var sonarQubeService = new Mock<ISonarQubeService>();
+
+            // Note: calling the public constructor here to check that it uses
+            // the expected set of languages
+            var testSubject = new QualityProfileDownloader(sonarQubeService.Object,
+                Mock.Of<IBindingConfigProvider>(),
+                Mock.Of<IConfigurationPersister>(),
+                Mock.Of<ISolutionBindingOperation>(),
+                new TestLogger(logToConsole: true));
+            
+            await testSubject.UpdateAsync(boundProject, null, CancellationToken.None);
+
+            foreach (var langauge in Language.KnownLanguages)
+            {
+                CheckLanguageQpRequested(sonarQubeService, langauge);
+            }
+            sonarQubeService.Invocations.Should().HaveCount(Language.KnownLanguages.Count());
+        }
+
+        [TestMethod]
+        public async Task UpdateAsync_MultipleQPs_ProgressEventsAreRaised()
```

Comment:
The checks in this test were previously included in `UpdateAsync_WhenQualityProfileIsNotAvailable_OtherLanguagesDownloadedSuccessfully` i.e. it was checking too much. I separated it into another test.